t/more-bundle.t

```diff
@@ -0,0 +1,264 @@
+            type => 'string',
+        },
+        '$refs which are simply $refs themselves are traversed automatically during resolution, at the top level too',
+    ],
+
+    [
+        'i_contain_refs_to_same_named_definitions',
+        {
+            definitions => code(sub {
+                my $got = shift;
+                return (0, 'expected hash with 2 keys')
+                    unless ref($got) eq 'HASH' and keys %$got == 2;
+                return (0, 'missing "dupe_name" key') if not exists $got->{dupe_name};
+                # we don't know which ref will keep its name and which will be renamed
+                my ($other_key) = grep $_ ne 'dupe_name', keys %$got;
+                return 1 if
+                    (eq_deeply($got->{dupe_name}, { type => 'integer' })
+                     and eq_deeply($got->{$other_key}, { type => 'string' })
+                     and eq_deeply($other_key, re(qr/-more-bundle2_yaml$/)))
+                 or ((eq_deeply($got->{dupe_name}, { type => 'string' })
+                     and eq_deeply($got->{$other_key}, { type => 'integer' })
+                     and eq_deeply($other_key, re(qr/-more-bundle_yaml$/))));
+                return (0, 'uh oh, got: '.(Test::More::explain($got))[0]);
+            }),
+            # begin i_contain_refs_to_same_named_definitions definition
+            type => 'object',
+            properties => {
+                foo => { '$ref' => re(qr/^#\/definitions\/(dupe_name|\w+-more-bundle_yaml)$/) },
+                bar => { '$ref' => re(qr/^#\/definitions\/(dupe_name|\w+-more-bundle2_yaml)/) },
+            },
+        },
+        'when encountering references that have the same root name, one is renamed',
+    ],
+
+    [
+        'i_have_a_ref_with_the_same_name',
+        {
+            definitions => {
+                i_have_a_ref_with_the_same_name => { type => 'string' },
+            },
+            # begin i_have_a_ref_with_the_same_name definition
+            type => 'object',
+            properties => {
+                name => { type => 'string' },
+                children => {
+                    type => 'array',
+                    items => { '$ref' => '#/definitions/i_have_a_ref_with_the_same_name' },
+                    default => [],
+                },
+            },
+        },
+        'we can handle pulling in references that have the same root name as the top level name',
+    ],
+
+    [
+        'i_am_a_ref_to_another_file',
+        {
+            definitions => {
+                ref3 => { type => 'integer' },
+            },
+            # begin i_am_a_ref_to_another_file definition - which is actually i_have_a_ref_to_the_first_filename
+            type => 'object',
+            properties => {
+                gotcha => { '$ref' => '#/definitions/ref3' },
+            },
+        },
+        'find and resolve a reference that immediately leaps to another file',
+    ],
+);
+
+my $_validator = JSON::Validator->new;
```

Comment:
I find it very confusing following $jv, $_jv and $_validator. Can we name them something more descriptive?

Reply:
I'll rename the variables, to indicate what they're each used for.